### PR TITLE
Update ui to 0.37.0

### PR DESCRIPTION
**Pull request by StephenWeatherford**

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
@dependabot recreate

File: .vscode/extensions.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "recommendations": [
-        "ms-vscode.vscode-typescript-tslint-plugin"
+        "ms-vscode.vscode-typescript-tslint-plugin",
+        "ms-vscode.extension-test-runner"
```
Comment (StephenWeatherford) [inline .vscode/extensions.json:4]: Extensions vscode will recommend for the project

File: .vscode/launch.json [hunk 2]
```diff
@@ -94,15 +92,44 @@
...
-            "stopOnEntry": false,
             "sourceMaps": true,
             // outFiles is used for locating generated JavaScript files, see https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_source-maps
             "outFiles": [
                 "${workspaceFolder}/dist/**/*.js"
             ],
             "preLaunchTask": "npm: pretest-dev",
             "env": {
-                "MOCHA_grep": "", // RegExp of tests to run (empty for all)
+                "MOCHA_grep": "", // RegExp of tests to run (empty for all) -- affects F5 only asdfg?
```
Comment (StephenWeatherford) [inline .vscode/launch.json:102]: Added new debugging option "Launch Tests With Webpack (release)"

File: gulpfile.ts [hunk 3]
```diff
@@ -129,11 +128,6 @@ async function test(): Promise<void> {
     if (result.error) {
       process.exit(1);
     }
-
```
Comment (StephenWeatherford) [inline gulpfile.ts:132]: This used to run the suites from here, now it just performs preparation for suites and they're actually run from elsewhere

File: package.json [hunk 4]
```diff
@@ -620,19 +620,24 @@
...
+        "webpack-prod": "npm run build && gulp webpack-prod",
         "webpack-profile": "webpack --profile --json --mode production > webpack-stats.json && echo Use http://webpack.github.io/analyse to analyze the stats",
         "build-grammars": "gulp build-grammars",
         "watch-grammars": "gulp watch-grammars",
         "verify-test-uses-extension-bundle": "gulp verify-test-uses-extension-bundle"
     },
+    "$devDependencies.comment": "@types/estree is to fix eslint issue, see https://stackoverflow.com/questions/67806963/types-eslint-error-chainexpression-and-importexpression-members-of-estree-not#:~:text=Solution%20to%20this%20problem%20is%3A%20npm%20install%20--save-dev,again%3A%20npm%20install%20Errors%20should%20be%20gone%20now.",
     "devDependencies": {
         "@types/decompress": "^4.2.3",
+        "@types/estree": "^1.0.1",
```
Comment (StephenWeatherford) [inline package.json:640]: Actual updated component versions...

File: package.json [hunk 5]
```diff
@@ -620,19 +620,24 @@
         "lint": "",
         "$lint-fix": "tslint --project tsconfig.json -t verbose --fix",
         "package": "npm run webpack-prod && npm run get-language-server && gulp package",
-        "pretest": "npm run build && npm run webpack-prod",
-        "pretest-dev": "npm run build && npm run webpack",
-        "test": "gulp test",
-        "all": "npm i && npm test && npm    run lint",
-        "webpack": "gulp webpack-dev",
-        "webpack-prod": "gulp webpack-prod",
+        "pretest": "npm run pretest-prod",
```
Comment (StephenWeatherford) [inline package.json:623]: A bit of cleanup mostly, plus fixing this line:

"test": "node ./out/test/runTest.js",
=>
"test": "node ./dist/test/runTest.js",

Which ran the tests in two contexts causing weird and confusing errors.

File: test/index.ts [hunk 6]
```diff
@@ -55,3 +60,22 @@ function addEnvVarsToMochaOptions(options: Mocha.MochaOptions): void {
         }
     }
 }
+
+if (isWebpack) {
```
Comment (StephenWeatherford) [inline test/index.ts:64]: Will help alert if we find ourselves in this situation again with picking up test files from the wrong location

File: test/support/getActionContext.ts [hunk 7]
```diff
@@ -1,18 +1,32 @@
-import { IActionContext } from "vscode-azureextensionui";
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
```
Comment (StephenWeatherford) [inline test/support/getActionContext.ts:5]: react to component breaking changes

File: package.json [hunk 8]
```diff
@@ -642,31 +647,31 @@
...
+        "mocha-multi-reporters": "^1.5.1",
         "nuget": "^2.0.2",
         "recursive-readdir": "^2.2.3",
         "rimraf": "3.0.2",
         "shelljs": "^0.8.5",
         "source-map-support": "^0.5.21",
         "ts-node": "^10.9.1",
-        "typescript": "^4.5.5",
-        "vsce": "^1.85.0",
+        "typescript": "^3.8.3",
```
Comment (jfleisher) [inline package.json:664]: Unusual to see a change to lower version. Assume this is intentional.

**Review by jfleisher — APPROVED** [review]

File: package.json (hunk 8 above)
Comment (StephenWeatherford) [reply package.json:664]: Yes, had to do that temporarily.  Will upgrade it again later.  Thanks!

**Pull request merged**